tests/js/ESLintBearTest.py

```diff
@@ -110,3 +184,49 @@ def test_lint_config_file(self):
             create_tempfile=False,
             filename=config_filename,
         )
+
+
+@generate_skip_decorator(ESLintBear)
+class ESLintBearLanguageTest(LocalBearTestHelper):
+
+    def setUp(self):
+        self.section = Section('name')
+        self.queue = Queue()
+        self.uut = ESLintBear(self.section, self.queue)
+
+    def test_markdown(self):
+        good_file = [line + '\n' for line in test_file_markdown.splitlines()]
+        bad_file = [line + '\n' for line in
+                    test_file_markdown_bad.splitlines()]
+        self.section.language = Language['Markdown']
+        self.check_validity(self.uut, good_file,
+                            tempfile_kwargs={'suffix': '.md'})
+        self.check_invalidity(self.uut, bad_file,
+                              tempfile_kwargs={'suffix': '.md'})
+
+    def test_HTML(self):
+        good_file = [line + '\n' for line in test_file_HTML.splitlines()]
+        bad_file = [line + '\n' for line in test_file_HTML_bad.splitlines()]
+        self.section.language = Language['html']
+        self.check_validity(self.uut, good_file,
+                            tempfile_kwargs={'suffix': '.html'})
+        self.check_invalidity(self.uut, bad_file,
+                              tempfile_kwargs={'suffix': '.html'})
+
+    def test_typscript(self):
+        good_file = [line + '\n' for line in test_file_typescript.splitlines()]
+        self.section.language = Language['Typescript']
+        self.check_validity(self.uut, good_file)
+
+    def test_bad_language(self):
+        test_file = [line + '\n' for line in test_good.splitlines()]
+        self.section.language = Language['Python']
+        self.check_validity(self.uut, test_file)
+        while not self.queue.empty():
+            message = self.queue.get()
+            msg = ('Language needs to be either Markdown, HTML, Typescript '
+                   'or Javascript. Assuming Javascript.')
+            if message.message == msg:
+                break
+        else:
+            assert False, 'Message not found'
```

Comment:
this one is currently failing.

to help with debugging, and also good for if it ever breaks, collect all of the messages encountered, and include them in the assertion message here.

or, collect all of the messages encountered, and use

`self.assertIn(expected_message, messages_encountered)`

That may help us see what went wrong.